tmt/base.py

```diff
@@ -1193,7 +1218,23 @@ def _save_tree(self, tree):
     def environment(self):
         """ Return environment combined from wake up and command line """
         combined = self._environment.copy()
-        combined.update(tmt.utils.environment_to_dict(self.opt('environment')))
+        environment_vars = tmt.utils.environment_to_dict(
+            self.opt('environment'))
+        environment_file_vars = tmt.utils.environment_file_to_dict(
+            self.opt('environment-file')
+            )
+
+        # check if environments data are not conflicitng and update the
+        # combined
+        assert not set(environment_vars).intersection(set(environment_file_vars)), (
```

Comment:
Should raise `tmt.utils.GeneralError` instead (see first comment about asserts)

Reply:
We've discussed this a bit on today's hacking session and the proposal is: Instead of raising errors on variable conflicts it would be better to define and document the precedence in which the `environment` and `environment-file` keys are evaluated. The proposed order is to check the `environment-file` first and then override with `environment`. This is similar to overriding the default config (in a file) with a command line option (explicit environment).

Reply:
resolved with the environment precedence over environment-file